pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -212,9 +292,15 @@ func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster
 		case !clusterAutoscalerUsed:
 			replicas = deployment.Minimum
 		// If the machine deployment does not yet exist we set replicas to min so that the cluster
-		// autoscaler can scale them as required.
+		// autoscaler can scale them as required. But if the machine deployment has some state records
+		// it means that we aither have controll plane migration or someone  intentionally has deleted
+		// the machine deployment
 		case existingMachineDeployment == nil:
-			replicas = deployment.Minimum
+			if deployment.State == nil {
+				replicas = deployment.Minimum
+			} else {
+				replicas = len(deployment.State.Machines)
```

Comment:
I don't understand this code - can you elaborate?

Reply:
Case hit question: `existingMachineDeployment == nil`
Does the wanted deployment we are about to deploy already exists. 
If not, we have to deploy it. But how many replicas have to have this deployment?
If this deployment was deleted(not by the worker controller) or we are migrating to another seed then we have a certain number of replicas. So we have to set the exact number or we are risking to let MCM delete some of the existing `Machines`. 
Does the wantedDeployment has state entry?: `deployment.State == nil`
If no, this means that this deployment have never existed before and we use the old approach to set replicas to `deployment.Minimum`. The ClusterAutoScaler will take care of the rest.
But If  the wantedDeployment has state entry, this mean that this deployment was existing recently and it is either manual deleted or we are migrating the control plane. 
How much Machine does it have? ` len(deployment.State.Machines)`

Reply:
Ah, okay, I see, thanks for the clarification, it makes sense after reading it again. :)